src/main/scala/mesosphere/marathon/state/GroupManager.scala

```diff
@@ -249,7 +252,15 @@ class GroupManager @Inject() (
         pms <- d.portMappings
       } yield {
         val mappings = pms.zip(servicePorts).map {
-          case (pm, sp) => pm.copy(servicePort = sp)
+          case (pm, sp) => pm match {
+            // merge service ports and container port mappings:
+            // always assign service port; if hostPort == 0, assign that to service port too
+            // (same affect as mergeServicePortsDefinitions, except on portMappings)
+            case Container.Docker.PortMapping(_, Some(0), 0, _, _, _) =>
```

Comment:
@gkleiman @aquamatthias I believe this is consistent with prior (pre-vnetworking) behavior. Please say something now if you think this will break the world.